### ISSUE TITLE
Potential Mercury Bug

### ISSUE THREAD

**Issue by pierrebeaucamp** [Masterminds/vcs#84]
Hi,

A tool I'm using is looking up some repository information using this library. I kept on getting a "The remote does not match the VCS endpoint" error so I investigated a bit. 

Turns out that this library is getting called as follows: `repo, err := vcs.NewRepo("", repoRoot)`. Notice the empty remote address which works fine for Git repositories but it seems to fail for Mercurial.

I think this is a bug in https://github.com/Masterminds/vcs/blob/master/hg.go#L47:
```
m := hgDetectURL.FindStringSubmatch(string(out))
if m[1] != "" && m[1] != remote {
        return nil, ErrWrongRemote
}
```

should be instead:
```
m := hgDetectURL.FindStringSubmatch(string(out))
if remote != "" && m[1] != remote {
        return nil, ErrWrongRemote
}
```

I just wanted to ask if this is intended behaviour or a bug. In case this is a bug, I'd be happy to open a PR.

**Comment by mattfarina** [comment]
@pierrebeaucamp You are right that this is a bug. If you want to file a PR I'd be happy to review.

Sorry for the slow response. I was a little underwater on a couple other things.

**Comment by pierrebeaucamp** [comment]
I'll give it a swing this weekend
